examples/soap-calculator/test/helpers.ts

```diff
@@ -0,0 +1,31 @@
+import {CalculatorDSDataSource} from '../src/datasources/calculator-ds.datasource';
+import {CalculatorServiceProvider} from '../src/services/calculator.service';
+
+// tslint:disable-next-line:no-any
+export async function givenAConnectedDataSource(): Promise<any> {
+  const calculatorDataSource = new CalculatorDSDataSource();
+
+  return new Promise(async (resolve, reject) => {
```

Comment:
What is the purpose of this Promise wrapper? IMO, this helper can be simplified as follows:

```ts
export async function givenAConnectedDataSource(): Promise<any> {
  const calculatorDataSource = new CalculatorDSDataSource();
  await calculatorDataSource.connect();
  return calculatorDataSource;
}
```

Unless I am missing something?

Reply:
I am working on these changes now. The wrapper was to help the integration test not to fail due to the fact that the connect method was not async at this time. Once PR #1563 is merged then your proposed code will work.